packages/x-date-pickers/src/DateCalendar/DateCalendar.tsx

```diff
@@ -449,17 +449,17 @@ export const DateCalendar = React.forwardRef(function DateCalendar<TDate>(
 
   const prevOpenViewRef = React.useRef(openView);
   React.useEffect(() => {
-    if (openView && openView !== focusedView) {
-      handleFocusedViewChange(openView)(true);
-    }
-
-    // Set focus to the button when switching from a view to another
+    // If the view change and the focus was on the previouse view
+    // Then we update the focus.
     if (prevOpenViewRef.current === openView) {
       return;
     }
+
+    if (focusedView === prevOpenViewRef.current && openView !== focusedView) {
```

Comment:
We can remove the 2nd check on this line.

If `focusedView === prevOpenViewRef.current`, then `openView !== focusedView`
Otherwise, we would have `openView === focusedView === prevOpenViewRef.current` 
And that's not possible given the check above (line 454)